crates/noirc_evaluator/src/errors.rs

```diff
@@ -8,95 +8,105 @@
-    UndeclaredAcirVar { location: Option<Location> },
+    UndeclaredAcirVar { call_stack: CallStack },
     #[error("ICE: Expected {expected:?}, found {found:?}")]
-    UnExpected { expected: String, found: String, location: Option<Location> },
+    UnExpected { expected: String, found: String, call_stack: CallStack },
 }
 
 impl RuntimeError {
-    fn location(&self) -> Option<Location> {
+    fn call_stack(&self) -> &CallStack {
         match self {
             RuntimeError::InternalError(
-                InternalError::DegreeNotReduced { location }
-                | InternalError::EmptyArray { location }
-                | InternalError::General { location, .. }
-                | InternalError::MissingArg { location, .. }
-                | InternalError::NotAConstant { location, .. }
-                | InternalError::UndeclaredAcirVar { location }
-                | InternalError::UnExpected { location, .. },
+                InternalError::DegreeNotReduced { call_stack }
+                | InternalError::EmptyArray { call_stack }
+                | InternalError::General { call_stack, .. }
+                | InternalError::MissingArg { call_stack, .. }
+                | InternalError::NotAConstant { call_stack, .. }
+                | InternalError::UndeclaredAcirVar { call_stack }
+                | InternalError::UnExpected { call_stack, .. },
             )
-            | RuntimeError::FailedConstraint { location, .. }
-            | RuntimeError::IndexOutOfBounds { location, .. }
-            | RuntimeError::InvalidRangeConstraint { location, .. }
-            | RuntimeError::TypeConversion { location, .. }
-            | RuntimeError::UnInitialized { location, .. }
-            | RuntimeError::UnknownLoopBound { location }
-            | RuntimeError::AssertConstantFailed { location }
-            | RuntimeError::UnsupportedIntegerSize { location, .. } => *location,
+            | RuntimeError::FailedConstraint { call_stack, .. }
+            | RuntimeError::IndexOutOfBounds { call_stack, .. }
+            | RuntimeError::InvalidRangeConstraint { call_stack, .. }
+            | RuntimeError::TypeConversion { call_stack, .. }
+            | RuntimeError::UnInitialized { call_stack, .. }
+            | RuntimeError::UnknownLoopBound { call_stack }
+            | RuntimeError::AssertConstantFailed { call_stack }
+            | RuntimeError::UnsupportedIntegerSize { call_stack, .. } => call_stack,
         }
     }
 }
 
 impl From<RuntimeError> for FileDiagnostic {
-    fn from(error: RuntimeError) -> Self {
-        let file_id = error.location().map(|loc| loc.file).unwrap();
-        FileDiagnostic { file_id, diagnostic: error.into() }
+    fn from(error: RuntimeError) -> FileDiagnostic {
+        let call_stack = vecmap(error.call_stack(), |location| *location);
+        let diagnostic = error.into_diagnostic();
+        let file_id = call_stack.last().map(|location| location.file).unwrap_or_default();
```

Comment:
So that in the unexpected case of an error without a span we still report the message even if it is to a default file. Rather than panicking entirely.